cwl/src/test/scala/wdl4s/cwl/CwlWorkflowWomSpec.scala

```diff
@@ -96,6 +98,64 @@ class CwlWorkflowWomSpec extends FlatSpec with Matchers {
     }
   }
 
+  behavior of "A decoded CWL 3step"
+
+  /**
+    * Runtime version of `refineMV[ECMAScriptExpression]("$(foo)")` as I cannot summon the dark magic for compile
+    * time checks.
+    *
+    * Compiler output after `import eu.timepit.refined._`:
+    *
+    * Error:(_, _) could not find implicit value for parameter v: eu.timepit.refined.api.Validate[String,wdl4s.cwl.ECMAScriptExpression]
+    * val foo = refineMV[ECMAScriptExpression]("$(foo)")
+    */
+  private def toECMAScriptExpression(unrefined: String): ECMAScriptExpression = {
+    eu.timepit.refined.api.RefType.applyRef[ECMAScriptExpression](unrefined).right
+      .getOrElse(fail(s"$unrefined was not an ECMAScriptExpression"))
+  }
+
+  private val stringOrExpressionTests = Table(
+    ("index", "result"),
+    (0, Inr(Inl("grep"))),
+    (1, Inl(toECMAScriptExpression("$(inputs.pattern)"))),
```

Comment:
ToL these look like low-level declarations of Coproducts?  I think you can do `Coproduct[StringOrExpression]("grep")`.  Arguably easier to read/maintain.